geomesa-fs/geomesa-fs-tools/src/main/scala/org/locationtech/geomesa/fs/tools/compact/ParquetCompactionJob.scala

```diff
@@ -0,0 +1,298 @@
+    job.setInputFormatClass(classOf[PartitionInputFormat])
+    job.setMapOutputKeyClass(classOf[Void])
+    job.setMapOutputValueClass(classOf[SimpleFeature])
+    ParquetInputFormat.setReadSupportClass(job, classOf[SimpleFeatureReadSupport])
+    PartitionInputFormat.setFsPath(job.getConfiguration, FileSystemDataStoreParams.PathParam.lookUp(dsParams).asInstanceOf[String])
+    PartitionInputFormat.setFsEncoding(job.getConfiguration, FileSystemDataStoreParams.EncodingParam.lookUp(dsParams).asInstanceOf[String])
+    PartitionInputFormat.setPartitions(job.getConfiguration, partitions.toArray)
+
+    // No reducers - Mapper will read/write its own things
+    job.setNumReduceTasks(0)
+
+    // Output format
+    job.setOutputFormatClass(classOf[SchemeOutputFormat])
+    SchemeOutputFormat.setFileType(job.getConfiguration, FileType.Compacted)
+    job.setOutputKeyClass(classOf[Void])
+    job.setOutputValueClass(classOf[SimpleFeature])
+    FileOutputFormat.setOutputPath(job, tempPath.getOrElse(dsPath))
+
+    // Parquet Options
+    val summaryLevel = Option(sft.getUserData.get(ParquetOutputFormat.JOB_SUMMARY_LEVEL).asInstanceOf[String])
+      .getOrElse(ParquetOutputFormat.JobSummaryLevel.NONE.toString)
+    job.getConfiguration.set(ParquetOutputFormat.JOB_SUMMARY_LEVEL, summaryLevel)
+    Command.user.info(s"Parquet metadata summary level is $summaryLevel")
+
+    val compression = Option(sft.getUserData.get(ParquetOutputFormat.COMPRESSION).asInstanceOf[String])
+      .map(CompressionCodecName.valueOf)
+      .getOrElse(CompressionCodecName.SNAPPY)
+    ParquetOutputFormat.setCompression(job, compression)
+    Command.user.info(s"Parquet compression is $compression")
+
+    // More Parquet config
+    ParquetOutputFormat.setWriteSupportClass(job, classOf[SimpleFeatureWriteSupport])
+    ParquetJobUtils.setSimpleFeatureType(job.getConfiguration, sft)
+
+    // Save the existing files so we can delete them afterwards
+    // Be sure to filter this based on the input partitions
+    val existingDataFiles = ds.storage.listPartitions(typeName)
+      .intersect(partitions).flatMap(ds.storage.getPaths(typeName, _)).toList
+
+    Command.user.info("Submitting job - please wait...")
+    job.submit()
+    Command.user.info(s"Tracking available at ${job.getStatus.getTrackingUrl}")
+
+    def mapCounters = Seq(("mapped", written(job)), ("failed", failed(job)))
+
+    val stageCount = if (tempPath.isDefined) { "2" } else { "1" }
+
+    while (!job.isComplete) {
+      if (job.getStatus.getState != JobStatus.State.PREP) {
+        val mapProgress = job.mapProgress()
+        if (mapProgress < 1f) {
+          statusCallback(s"Map (stage 1/$stageCount): ", mapProgress, mapCounters, done = false)
+        } else {
+          statusCallback(s"Map (stage 1/$stageCount): ", mapProgress, mapCounters, done = true)
+          statusCallback.reset()
+        }
+      }
+      Thread.sleep(1000)
+    }
+
+    val res = (written(job), failed(job))
```

Comment:
[minor] `ret` and `res` are pretty similar names, maybe get rid of res and just return this at the end?